include/boost/geometry/algorithms/detail/is_valid/has_invalid_coordinate.hpp

```diff
@@ -80,26 +80,18 @@ struct indexed_has_invalid_coordinate
 
 struct range_has_invalid_coordinate
 {
-    struct point_has_valid_coordinates
-    {
-        template <typename Point>
-        static inline bool apply(Point const& point)
-        {
-            return ! point_has_invalid_coordinate::apply(point);
-        }
-    };
-
     template <typename Geometry, typename VisitPolicy>
     static inline bool apply(Geometry const& geometry, VisitPolicy& visitor)
     {
         boost::ignore_unused(visitor);
 
-        bool const has_valid_coordinates = detail::check_iterator_range
-            <
-                point_has_valid_coordinates,
-                true // do not consider an empty range as problematic
-            >::apply(geometry::points_begin(geometry),
-                     geometry::points_end(geometry));
+        auto const points_end = geometry::points_end(geometry);
+        bool const has_valid_coordinates = std::find_if
+            (
+                geometry::points_begin(geometry), points_end,
+                []( auto const& point ){ return point_has_invalid_coordinate::
+                    apply(point); }
```

Comment:
Inconsistent formatting of lambdas:
```
[]( auto const& point ) {
    return point_has_invalid_coordinate::apply(point);
}
```

Or you could add `point_has_invalid_coordinate::operator()` and simply pass it here.

Reply:
Thanks, I changed the format.